specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.4-preview.1/settings.json

```diff
@@ -0,0 +1,208 @@
+          "description": "Retrieves the value of a specified, value account setting.",
+          "parameters": [
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "An account setting containing the name of the setting and the value it is set to.",
+              "schema": {
+                "$ref": "#/definitions/Setting"
+              }
+            },
+            "default": {
+              "description": "Key Vault error response describing why the operation failed.",
+              "schema": {
+                "$ref": "common.json#/definitions/KeyVaultError"
+              }
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextLink"
+          },
+          "x-ms-examples": {
+            "GetKeys": {
+              "$ref": "./examples/UpdateSetting-example.json"
+            }
+          }
+        }
+      },
+      "/settings": {
+        "get": {
+          "tags": [
+            "Settings"
+          ],
+          "operationId": "GetSettings",
+          "summary": "List account settings.",
+          "description": "Retrieves a list of all the available account settings that can be configured.",
+          "parameters": [
+            {
+              "$ref": "#/parameters/ApiVersionParameter"
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "A response message containing a list of account settings with their associated value.",
+              "schema": {
+                "$ref": "#/definitions/SettingsListResult"
+              }
+            },
+            "default": {
+              "description": "Key Vault error response describing why the operation failed.",
+              "schema": {
+                "$ref": "common.json#/definitions/KeyVaultError"
+              }
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextLink"
+          },
+          "x-ms-examples": {
+            "GetKeys": {
+              "$ref": "./examples/ListSettings-example.json"
+            }
+          }
+        }
+      }
+    },
+    "definitions": {
+      "UpdateSettingsRequest": {
+        "properties": {
+          "value": {
+            "type": "string",
+            "description": "The account setting value. True if setting is enabled, false if disabled"
+          }
+        },
+        "description": "The update settings request object.",
+        "required": [
+          "value"
+        ]
+      },
+      "Setting": {
+        "properties": {
+          "name": {
+            "type": "string",
+            "description": "The account setting to be updated"
+          },
+          "value": {
+            "type": "string",
+            "description": "The account setting value. True if setting is enabled, false if disabled"
```

Comment:
Why is this defined as a string if you're describing boolean values?

Reply:
Right now, we only have one setting that only needs to be Off/On or in other words True/False. In the future, we will likely need settings that can hold string values and not be limited to true or false. I will update this description though to not limit it to only a "boolean" type definition.

Reply:
Changing the `type` to something like `object` is not going to make for a good API. Off hand, I would say this needs to be a polymorphic type but having derivatives with just a single typed value seems overkill. Please set up a meeting - or come to office hours (for something this small, that's probably better) - with the API Stewardship board.

Reply:
Alternatively, look at AppConfiguration: https://github.com/Azure/azure-rest-api-specs/blob/4b5fe2fb0a5066c4ff2bd429dbd5e1afda6afab3/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/appconfiguration.json#L1194-L1226

That said, this still isn't a great experience for client libraries since strong-typed languages will need to provide an abstraction over the type when we're trying to move to more generated code by default.

Reply:
Office hours discussion: Agreed on adding a type specifier to support multiple intrinsic types 

```
"value": {
  "type": "string"
},
"type": {
  "enum": [
    "boolean",
    "string",
    "integer",
    "float"
  ]
}
```